api/checks/runs.go

```diff
@@ -42,12 +113,11 @@ func updateCheckRun(ctx context.Context, summary summaries.Summary, suites ...sh
 		opts.CompletedAt = &github.Timestamp{Time: time.Now()}
 	}
 
-	for _, suite := range suites {
-		created, err := createCheckRun(ctx, suite, opts)
-		if !created || err != nil {
-			return false, err
-		}
-		log.Debugf("Check for %s/%s @ %s (%s) updated", suite.Owner, suite.Repo, suite.SHA[:7], state.Product.String())
+	client, err := getGitHubClient(ctx, suite.AppID, suite.InstallationID)
+	if err != nil {
+		return false, err
 	}
-	return true, nil
+
+	_, _, err = client.Checks.UpdateCheckRun(ctx, suite.Owner, suite.Repo, run.GetID(), opts)
```

Comment:
It might be good to extract this part into `updateCheckRun` and place it next to `createCheckRun`, which is currently in `api/checks/webhook.go` (you may also move them together into `api.go`).

Reply:
I'll follow "rules of three" on this one, and do it when I get another caller.